From 81f0e7de9eb1cb30806dd44c615d2cc895b58a49 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sat, 9 Feb 2019 19:16:15 +0100
Subject: [PATCH] texture-cache: Use content size for returned images

Since commit deec0bf2550, the texture cache is based on ClutterImage
rather than ClutterTexture. As ClutterImage (like all ClutterContent)
is only concerned with painting, it doesn't influence the size of the
actor it is added to at all, and the returned actor will now stay at
size 0x0 after the image has been loaded.

Set up the actor to follow the content's size instead, to get closer
to the previous behavior.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/402
---
 src/st/st-texture-cache.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/st/st-texture-cache.c b/src/st/st-texture-cache.c
index 1d11ce71c..967ad9e2e 100644
--- a/src/st/st-texture-cache.c
+++ b/src/st/st-texture-cache.c
@@ -71,6 +71,7 @@ create_invisible_actor (void)
 {
   return g_object_new (CLUTTER_TYPE_ACTOR,
                        "opacity", 0,
+                       "request-mode", CLUTTER_REQUEST_CONTENT_SIZE,
                        NULL);
 }
 
-- 
2.20.1

